Ical.Net/Calendar.cs

```diff
@@ -238,11 +244,17 @@ public virtual HashSet<Occurrence> GetOccurrences(DateTime startTime, DateTime e
     /// <param name="dt">The date for which to return occurrences. Time is ignored on this parameter.</param>
     /// <returns>A list of Periods representing the occurrences of this object.</returns>
     public virtual HashSet<Occurrence> GetOccurrences<T>(IDateTime dt) where T : IRecurringComponent
-        => GetOccurrences<T>(new CalDateTime(dt.Date), new CalDateTime(dt.Date.AddDays(1).AddTicks(-1)));
+    {
+        var endTimeOnly = dt.Time?.Add(TimeSpan.FromTicks(-1));
+        return GetOccurrences<T>(new CalDateTime(dt.Date, dt.Time), new CalDateTime(dt.Date.AddDays(1), endTimeOnly));
+    }
 
     /// <inheritdoc cref="GetOccurrences(IDateTime)"/>
     public virtual HashSet<Occurrence> GetOccurrences<T>(DateTime dt) where T : IRecurringComponent
-        => GetOccurrences<T>(new CalDateTime(dt.Date), new CalDateTime(dt.Date.AddDays(1).AddTicks(-1)));
```

Comment:
This is probably meant to be `TimeSpan.FromSeconds(-1)` because 1 tick just gets truncated?